plugins/modules/proxmox_pool_member.py

```diff
@@ -0,0 +1,240 @@
+
+EXAMPLES = """
+- name: Add new VM to Proxmox VE pool
+  community.general.proxmox_pool_member:
+    api_host: node1
+    api_user: root@pam
+    api_password: password
+    poolid: test
+    member: 101
+
+- name: Add new storage to Proxmox VE pool
+  community.general.proxmox_pool_member:
+    api_host: node1
+    api_user: root@pam
+    api_password: password
+    poolid: test
+    member: zfs-data
+    type: storage
+
+- name: Remove VM from the Proxmox VE pool using VM name
+  community.general.proxmox_pool_member:
+    api_host: node1
+    api_user: root@pam
+    api_password: password
+    poolid: test
+    member: pxe.home.arpa
+    state: absent
+
+- name: Remove storage from the Proxmox VE pool
+  community.general.proxmox_pool_member:
+    api_host: node1
+    api_user: root@pam
+    api_password: password
+    poolid: test
+    member: zfs-storage
+    type: storage
+    state: absent
+"""
+
+RETURN = """
+poolid:
+  description: The pool ID.
+  returned: success
+  type: str
+  sample: test
+member:
+  description: Member name.
+  returned: success
+  type: str
+  sample: 101
+msg:
+  description: A short message on what the module did.
+  returned: always
+  type: str
+  sample: "Member 101 deleted from the pool test"
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.proxmox import (proxmox_auth_argument_spec, ProxmoxAnsible)
+
+
+class ProxmoxPoolMemberAnsible(ProxmoxAnsible):
+
+    def pool_members(self, poolid):
+        vms = []
+        storage = []
+        for member in self.get_pool(poolid)["members"]:
+            if member["type"] == "storage":
+                storage.append(member["storage"])
+            else:
+                vms.append(member["vmid"])
+
+        return (vms, storage)
+
+    def add_pool_member(self, poolid, member, member_type):
+        current_vms_members, current_storage_members = self.pool_members(poolid)
+        all_members_before = current_storage_members + current_vms_members
+        all_members_after = all_members_before.copy()
+        diff = {"before": {"members": all_members_before}, "after": {"members": all_members_after}}
+
+        try:
+            if member_type == "storage":
+                storages = self.get_storages(type=None)
+                if member not in [storage["storage"] for storage in storages]:
+                    self.module.fail_json(msg="Storage {0} doesn't exist in the cluster".format(member))
+                if member in current_storage_members:
+                    self.module.exit_json(changed=False, poolid=poolid, member=member,
+                                          diff=diff, msg="Member {0} is already part of the pool {1}".format(member, poolid))
+
+                all_members_after.append(member)
+                if self.module.check_mode:
+                    return diff
+
+                self.proxmox_api.pools(poolid).put(storage=[member])
+                return diff
```

Comment:
You could also write this as:
```suggestion
                if not self.module.check_mode:
                    self.proxmox_api.pools(poolid).put(storage=[member])
                return diff
```
(same below)